lib/lib.go

```diff
@@ -0,0 +1,189 @@
+package lib
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/spf13/viper"
+)
+
+// Podium is a struct that represents a podium API application
+type Podium struct {
+	Config            *viper.Viper
+	URL               string
+	baseLeaderboard   string
+	localeLeaderboard string
+}
+
+// NewPodium returns a new podium API application
+func NewPodium(config *viper.Viper) *Podium {
+	app := &Podium{
+		Config:            config,
+		URL:               config.GetString("podium.url"),
+		baseLeaderboard:   config.GetString("leaderboards.globalLeaderboard"),
+		localeLeaderboard: config.GetString("leaderboards.localeLeaderboard"),
+	}
+	return app
+}
+
+func sendTo(method, url string, payload map[string]interface{}) (int, string, error) {
+	payloadJSON, err := json.Marshal(payload)
+	if err != nil {
+		return -1, "", err
+	}
+
+	var req *http.Request
+
+	if payload != nil {
+		req, err = http.NewRequest(method, url, bytes.NewBuffer(payloadJSON))
+		if err != nil {
+			return -1, "", err
+		}
+	} else {
+		req, err = http.NewRequest(method, url, nil)
+		if err != nil {
+			return -1, "", err
+		}
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{}
```

Comment:
instead of creating a new http.Client in every request it's better to share a single one and use keep-alive requests, see here: https://awmanoj.github.io/tech/2016/12/16/keep-alive-http-requests-in-golang/